### PR TITLE
[ALLI-7719] RecordDataFormatter: removed all the entries with no function in the driver

### PR DESCRIPTION
**Pull request by LuomaJuha**
The simple function used:

```php
var_dump(count($include));
echo "<pre>";
$intersected = array_intersect_key($coreFields, array_flip($include));
$ind = 0;
$realFunctions = [];
foreach ($intersected as $key => $value) {
    if (is_callable([$driver, $value['dataMethod']], true, $callable_name)) {
        if (method_exists($driver, $value['dataMethod'])) {
            $realFunctions[] = $key;
        }
    } else {
        // Sometimes the dataMethod is not a function name, so save it in case
        $realFunctions[] = $key;
    }
}
sort($realFunctions);
$j = 0;
for ($i = 0; $i < count($realFunctions); $i++) {
    echo "'{$realFunctions[$i]}', ";
    if (++$j === 3) {
        echo PHP_EOL;
        $j = 0;
    } 
}
echo "</pre>";
var_dump($ind);
```

### CODE REVIEW

**Comment by LuomaJuha** [comment]
Huoh, se funktio meni vaan yheks pötköks...

**Review by EreMaijala — CHANGES_REQUESTED** [review]
Let's just make it so that there's one field per line. It's longer, but it'll make making and diffing changes so much easier later on.

**Comment by LuomaJuha** [comment]
Added also an entry for lido in the function filterCollectionFields

**Review by EreMaijala — CHANGES_REQUESTED** [review]
Sorry, one more thing: I think it would be better to use case-insensitive sort. I find it confusing that e.g. `child_records` is at the bottom of the list.

**Comment by LuomaJuha** [comment]
Noticed that I didn't create a ticket for this one. Will do so the history is recorded somewhere.

File: module/Finna/src/Finna/View/Helper/Root/RecordDataFormatter.php [hunk 1]
```diff
@@ -503,56 +570,14 @@ public function filterForwardFields($coreFields)
...
-            'Unit ID', 'Unit IDs'
-        ];
-
-        $fields = array_intersect_key($coreFields, array_flip($include));
-
-        return $type === 'ead' ?
-            $this->filterEADFields($fields) :
-            $this->filterEAD3Fields($fields);
+        switch ($type) {
+        case 'ead':
```
Comment (EreMaijala) [inline module/Finna/src/Finna/View/Helper/Root/RecordDataFormatter.php:577]: Laitetaan EAD3 `case 'ead3':`:lle eikä tehdä defaultina mitään tai viskataan poikkeus tai jotain.

**Review by EreMaijala — CHANGES_REQUESTED** [review]

File: module/Finna/src/Finna/View/Helper/Root/RecordDataFormatter.php (hunk 1 above)
Comment (LuomaJuha) [reply module/Finna/src/Finna/View/Helper/Root/RecordDataFormatter.php:577]: jepulis!

Comment (LuomaJuha) [reply module/Finna/src/Finna/View/Helper/Root/RecordDataFormatter.php:577]: Laitoin pienen poikkeuksen tulemaan jos yrittää tietuetyypillä, mil ei oo tukea collectionille.

**Review by EreMaijala — APPROVED** [review]

**Pull request merged**